db/migrate/20161214033216_enable_uuid_extension.rb

```diff
@@ -0,0 +1,11 @@
+class EnableUuidExtension < ActiveRecord::Migration
+  def change
+    enable_extension 'uuid-ossp'
```

Comment:
This is definitely the "correct" way to do this. Unfortunately, I'm not confident that the rails user (`www-data`) in production has sufficient permissions to enable postgres extensions unilaterally. We could potentially enable it in puppet, or we could work around it in a slightly different manner.

You could use a normal integer primary key and then create a `text` type column for the UUID. Then in an activerecord callback in your `BadgeClaim` object (probably `after_create`) you could do something like

```ruby
after_create :init_uuid

def init_uuid
  unless self.redemption_code.present?
    self.redemption_code = SecureRandom.uuid
    self.save
  end
end
```